src/python/espressomd/observables.py

```diff
@@ -83,10 +82,10 @@ def __init__(self, **kwargs):
             try_vectors = [np.array([1., 0., 0.]), np.array([0., 0., 1.])]
             axis = np.asarray(kwargs["axis"])
             for vec in try_vectors:
-                proj = np.dot(vec, axis / np.linalg.norm(axis))
-                if np.arccos(proj) > 10 * sys.float_info.epsilon:
-                    vec -= proj * axis
-                    kwargs["orientation"] = vec / np.linalg.norm(vec)
+                orthogonal_component = vec - np.dot(vec, axis) * axis
+                norm = np.linalg.norm(orthogonal_component)
+                if norm > 0.01:
```

Comment:
what is this magic 0.01 here? do want to check for ```sys.float_info.epsilon```, maybe?

Reply:
why do we replicate the logic from the core here?

Reply:
1.) if the orthogonal component is 2 epsilon long and we scale it to 1, it will probably not be very orthogonal due to precision error. I chose a very generous value of 0.1 because we always have the second vector to try which will be more orthogonal.
2.) The core logic for default orientation is for hollowconicalfrustum, the python one for observables

Reply:
* okay
* i think the logic is in the wrong place then?

Reply:
Where would you suggest is the right place to unite it?

Reply:
algorithmically I think we could do something like: https://math.stackexchange.com/a/1681815

Reply:
but that is what I am doing here :)
The Gram-Schmidt-type orthogonalisation works in whatever dimension
You just have to start with a trial vector that is not parallel